### PR TITLE
Weinflash - Add FeedbackForm/Store to collect user name and comments

### PR DESCRIPTION
**Pull request by dweinflash**
Add a Feedback form for users to input a user name and comments. Include a submit button on the bottom of the Feedback form. Feedback store capable of storing user names and comments.

### CODE REVIEW

File: app/javascript/image_sharing/test/components/App.spec.js [hunk 1]
```diff
@@ -3,13 +3,23 @@ import { expect } from 'chai';
...
+  it('app should render components', () => {
+    const should = require('chai').should();
+    const feedbackStore = { feedbackStore: {} };
+    const wrapper = mount(<App stores={feedbackStore} />);
+
     const headerText = 'Tell us what you think';
+    const footerText = 'Copyright: AppFolio Inc. Onboarding';
+
     const header = <h3 className='text-center'>{headerText}</h3>
+    const footer = <p className='footer'>{footerText}</p>
```
Comment (wanyue0714) [inline app/javascript/image_sharing/test/components/App.spec.js:18]: There is an easier way to test header and footer:
First import these two components
```
import Header from '';
import Footer from '';
```
Then find this component in wrapper
```
const header = wrapper.find(Header);
const footer = wrapper.find(Footer);
```
Finally you just need expect it does find them
```
expect(header).to.have.length(1);
expect(footer).to.have.length(1);
```
So you don't need to build header and footer again

File: app/javascript/image_sharing/test/components/App.spec.js [hunk 2]
```diff
@@ -3,13 +3,23 @@ import { expect } from 'chai';
...
+    const form = wrapper.find(FeedbackForm);
+
     expect(wrapper.contains(header)).to.equal(true);
+    expect(wrapper.contains(footer)).to.equal(true);
+    should.exist(form);
```
Comment (wanyue0714) [inline app/javascript/image_sharing/test/components/App.spec.js:23]: Same here, it's okay to use should.exist here, but could be a little simpler as:
```
const form = wrapper.find(FeedbackForm);
expect(form).to.have.length(1);
```
This above already means we find a form, and it only be rendered once

File: app/javascript/image_sharing/test/components/FeedbackForm.spec.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import { mount } from 'enzyme';
+import { expect } from 'chai';
+import FeedbackForm from '../../components/FeedbackForm';
+import { Form, Button, Input } from 'reactstrap';
+
+describe('FeedbackForm Test', () => {
+  it('form should render components and store', () => {
```
Comment (wanyue0714) [inline app/javascript/image_sharing/test/components/FeedbackForm.spec.js:8]: Actually, store hasn't been used in form right? https://github.com/AppFolioOnboarding/image-sharer-dweinflash/pull/17/files#diff-6bc204456bb173edc0981da6fa363af0
So either we add it into FeedbackForm (display username and comments in the form), or we could skip store test in this story

File: app/javascript/image_sharing/test/components/FeedbackForm.spec.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    feedbackStore.userName = 'david.weinflash';
+    feedbackStore.comments = 'asdf';
+
+    const inputName = form.find(Input).at(0);
+    const inputComments = form.find(Input).at(1);
+
+    expect(inputName.props()).to.deep.equal({
+      className: 'js-username',
+      type: 'text'
+    });
```
Comment (wanyue0714) [inline app/javascript/image_sharing/test/components/FeedbackForm.spec.js:25]: If we display store value in `FeedbackForm `, we could test the value here, sth like:
```
expect(inputName.props()).to.deep.equal({
      className: 'js-username',
      type: 'text',
      value: 'david.weinflash'
});
```

File: app/javascript/image_sharing/test/components/FeedbackForm.spec.js (hunk 3 above)
Comment (dweinflash) [reply app/javascript/image_sharing/test/components/FeedbackForm.spec.js]: Oh yes you're right! I'll save the store tests for the next story, where the story is more in line with data storage and the PostFeedbackService

File: app/javascript/image_sharing/test/components/FeedbackForm.spec.js (hunk 4 above)
Comment (dweinflash) [reply app/javascript/image_sharing/test/components/FeedbackForm.spec.js:27]: In this story I'll test that the store is technically capable of storing data. In React Issue 3, I'll make sure that the user can effectively save data in the store using the FeedbackForm

**Pull request merged**